### PR TITLE
feat(gprogram): export rpc calls from Api

### LINKED ISSUE #1780: Add rpc-calls support for `gclient` from `GearApi`
### Problem to Solve

.

### Possible Solution

.

### Notes

_No response_

### PR DESCRIPTION
**Pull request by clearloop**
Resolves #1780 

- move rpc module from signer to api
-

@gear-tehc/dev

### CODE REVIEW

**Comment by SabaunT** [comment]
Can we see a couple of tests in 'gclient' of rpc calls?

File: program/src/api/rpc.rs [hunk 1]
```diff
@@ -35,7 +30,7 @@ impl Signer {
             .request(
                 "gear_calculateInitCreateGas",
                 rpc_params![
-                    self.source(),
+                    H256(Default::default()),
```
Comment (breathx) [inline program/src/api/rpc.rs:33]: why source removed?

Comment (breathx) [reply program/src/api/rpc.rs:33]: this thing may affect calculate gas logic

File: gclient/src/api/mod.rs [hunk 2]
```diff
@@ -90,3 +91,17 @@ impl GearApi {
             .map_err(Into::into)
     }
 }
+
+impl Deref for GearApi {
```
Comment (breathx) [inline gclient/src/api/mod.rs:95]: Do we really need all methods of Signer available for GearApi?

**Review by breathx — CHANGES_REQUESTED** [review]
Agree with @SabaunT, some tests are required just for demonstration
Otherwise lgtm!

File: program/src/api/rpc.rs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        at: Option<H256>,
+    ) -> Result<GasInfo> {
+        self.rpc()
+            .request(
+                "gear_calculateHandleGas",
+                rpc_params![
+                    origin,
+                    H256(destination.into()),
+                    hex::encode(payload),
+                    u64::try_from(value).unwrap_or(u64::MAX),
```
Comment (breathx) [inline program/src/api/rpc.rs:90]: do we really can't throw here u128? use [u8; 16] instead for rpc calls also

**Review by breathx — APPROVED** [review]
I'd make source argument optional and paste self id in case of none

**Comment by breathx** [comment]
Add and similar bindings for `gclient`, please!

File: program/src/api/rpc.rs (hunk 3 above)
Comment (breathx) [reply program/src/api/rpc.rs]: we never used u64 for value
If I'm not mistaken, rpc params macro wasn't able to use it for some reason

**Pull request merged**